### PR TITLE
Cleanup sonar issues and code duplications in PackedAtlas

### PR DESCRIPTION
**Pull request by matthieun**
### Description:

Cleanup sonar issues and code duplications in `PackedAtlas`.

### Potential Impact:

Should have no impact.

### Unit Test Approach:

Using same test coverage

### Test Results:

Ran large scale tests and everything seems all right, performance wise and data wise. This should be good to go.

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlas.java [hunk 1]
```diff
@@ -475,22 +480,8 @@ public Area area(final long identifier)
...
-                return this.index < PackedAtlas.this.areaIdentifiers().size();
-            }
-
-            @Override
-            public Area next()
-            {
-                return new PackedArea(PackedAtlas.this, this.index++);
-            }
-        };
+        return indexBasedIterable(this.areaIdentifiers().size(),
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlas.java]: 👍

File: src/main/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlas.java [hunk 2]
```diff
@@ -1503,199 +1364,131 @@ private void addRelationMember(final String type, final long relationIndex,
...
+        return deserializedIfNeeded(supplier, null, lock, fieldName);
+    }
+
+    private <T> T deserializedIfNeeded(final Supplier<T> supplier, final Consumer<T> consumer,
+            final Object lock, final String fieldName)
+    {
+        if (supplier.get() == null)
         {
-            synchronized (FIELD_AREA_TAGS_LOCK)
+            synchronized (lock) // NOSONAR
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlas.java]: 😱

**Comment by matthieun** [comment]
There are a lot of manual fixes in this PR, so if anyone can focus on searching for potential copy-paste typos, this would help!

File: src/main/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlas.java [hunk 3]
```diff
@@ -1461,7 +1333,7 @@ protected void setMetaData(final AtlasMetaData metaData)
     }
 
     @Override
-    protected void setName(final String name)
+    protected void setName(final String name) // NOSONAR
```
Comment (lucaspcram) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlas.java]: Sneaky sneaky :D

**Pull request merged**